clippy_lints/src/if_then_some_else_none.rs

```diff
@@ -70,21 +70,21 @@ impl<'tcx> LateLintPass<'tcx> for IfThenSomeElseNone {
             && let ExprKind::Block(then_block, _) = then.kind
             && let Some(then_expr) = then_block.expr
             && let ExprKind::Call(then_call, [then_arg]) = then_expr.kind
-            && let ctxt = expr.span.ctxt()
-            && then_expr.span.ctxt() == ctxt
+            && !expr.span.from_expansion()
+            && !then_expr.span.from_expansion()
             && is_res_lang_ctor(cx, path_res(cx, then_call), OptionSome)
             && is_res_lang_ctor(cx, path_res(cx, peel_blocks(els)), OptionNone)
             && !is_else_clause(cx.tcx, expr)
             && !is_in_const_context(cx)
-            && !expr.span.in_external_macro(cx.sess().source_map())
```

Comment:
Is there a reason for why we don't want to lint code from within local macros anymore?

Reply:
Good question. Is there a context in which I can get the non-expanded code? If the `if … { Some(…) } else { None }` is in the macro, using the context of the `if` expression is not enough to get non-expanded code for the condition or the content of the `Some`. How can I retrieve it?